src/eradius_client.erl

```diff
@@ -323,6 +382,31 @@ configure(State) ->
             {error, {bad_client_ip, ClientIP}}
     end.
 
+%% private
+prepare_pools() ->
+    ets:new(?MODULE, [ordered_set, public, named_table, {keypos, 1}, {write_concurrency,true}]),
+    lists:foreach(fun({_PoolName, Servers}) -> prepare_pool(Servers) end, application:get_env(eradius, servers_pool, [])).
+prepare_pool([]) -> ok;
+prepare_pool([{Addr, Port, _, Opts} | Servers]) ->
+    Retries = proplists:get_value(retries, Opts, ?DEFAULT_RETRIES),
+    store_radius_server_from_pool(Addr, Port, Retries),
+    prepare_pool(Servers);
+prepare_pool([{Addr, Port, _} | Servers]) ->
+    store_radius_server_from_pool(Addr, Port, ?DEFAULT_RETRIES),
+    prepare_pool(Servers).
+
+%% private
+store_radius_server_from_pool(Addr, Port, Retries) when is_tuple(Addr) ->
+    ets:insert(?MODULE, {{Addr, Port}, Retries, Retries});
+store_radius_server_from_pool(Addr, Port, Retries) ->
+    case inet:gethostbyname(Addr) of
```

Comment:
configuring RADIUS peers by name is usually a problem. That originates from RFC 2865, Sect. 3 that states:

> A RADIUS server MUST use the source IP address of the RADIUS UDP packet to decide which shared secret to use

Name based configuration hide the IP and might therefore lead to hard find misconfigurations of the shared secret.

Reply:
reworked this, it will throw error if not an IP address will be given.